_includes/markdown/JavaScript.md

```diff
@@ -304,6 +304,77 @@ document.getElementById( 'menu' ).addEventListener( 'click', ( e ) => {
 
 You may be wondering why we don't just add one listener to the `<body>` for all our events. Well, we want the event to *bubble up the DOM as little as possible* for [performance reasons](https://jsperf.com/event-delegation-distance). This would also be pretty messy code to write.
 
+### Debounce and Throttle
+
+Browser events such as scrolling, resizing, and cursor movements happen as fast as possible and can cause performance issues. Debouncing and throttling our functions helps us increase performance by controlling the rate at which an event listener calls them. 
+
+Debouncing a function will prevent it from being called again until a defined amount of time has passed, i.e., execute this function if 200ms has passed since it was last called. A common use case would be when resizing a browser window; we can apply classes or move elements after the resize has happened.
+
+Throttling a function will cause it to only be called a maximum number of times over a defined period of time, i.e., only execute this function once every 50ms. A common use case would be when scrolling a browser window; we may want visual effects as a scroll is happening.
+
+Here are example debounce and throttle functions, though note that some of our recommended utility libraries already have similar functions, such as Underscore's _.debounce() and _.throttle(). 
+
+```javascript
```

Comment:
Can you move this under the debounce paragraph so we don't have two code blocks next to eachother? Then add another note for something like, "this is a debounce example". Also it would be nice to get some comments in the code talking about what's actually going on here